plugins/greatexpectations/flytekitplugins/greatexpectations/task.py

```diff
@@ -0,0 +1,166 @@
+        except KeyError:
+            raise KeyError(f"Your kwargs key has to be {list(self.python_interface.inputs.keys())[0]}")
+
+        if type(dataset) != str:
+            raise RuntimeError(f"'dataset' has to have string data type")
+
+        final_batch_request = {
+            "data_asset_name": dataset,
+            "datasource_name": self._data_source,
+            "data_connector_name": self._data_connector,
+        }
+
+        # GE's RuntimeBatchRequest
+        if self._batch_request and self._batch_request.runtime_parameters:
+            final_batch_request.update(
+                {
+                    "runtime_parameters": self._batch_request.runtime_parameters,
+                    "batch_identifiers": self._batch_request.batch_identifiers,
+                    "batch_spec_passthrough": self._batch_request.batch_spec_passthrough,
+                }
+            )
+
+        # GE's BatchRequest
+        elif self._batch_request:
+            final_batch_request.update(
+                {
+                    "data_connector_query": self._batch_request.data_connector_query,
+                }
+            )
+
+        checkpoint_config = {
+            "class_name": "SimpleCheckpoint",
+            "validations": [
+                {
+                    "batch_request": final_batch_request,
+                    "expectation_suite_name": self._expectation_suite,
+                }
+            ],
+        }
+
+        if self._checkpoint_params:
+            checkpoint = SimpleCheckpoint(
+                f"_tmp_checkpoint_{self._expectation_suite}", context, **checkpoint_config, **self._checkpoint_params
+            )
+        else:
+            checkpoint = SimpleCheckpoint(f"_tmp_checkpoint_{self._expectation_suite}", context, **checkpoint_config)
+
+        # identify every run uniquely
+        run_id = RunIdentifier(
+            **{
+                "run_name": self._data_source + "_run",
+                "run_time": datetime.datetime.utcnow(),
+            }
+        )
+
+        checkpoint_result = checkpoint.run(run_id=run_id)
+        final_result = convert_to_json_serializable(checkpoint_result.list_validation_results())[0]
+
+        result_string = ""
+        if final_result["success"] is False:
+            for every_result in final_result["results"]:
+                if every_result["success"] is False:
+                    result_string += (
+                        every_result["expectation_config"]["kwargs"]["column"]
+                        + " -> "
+                        + every_result["expectation_config"]["expectation_type"]
+                        + "\n"
+                    )
+
+            # print column names for which the validation failed
+            print(f"\nCOLUMN\t\tFAILED EXPECTATION")
+            print(result_string)
+
+            # raise a GE exception
+            raise ValidationError(f"Validation failed!")
```

Comment:
Lets raise the validation error and add the entire failure information and `result_string` to the error message. that way. Flytekit will capture it and you will be able to see it in the UI